string location
string sub_location
---
bool success
